x-pack/plugins/infra/common/log_analysis/job_parameters.ts

```diff
@@ -6,21 +6,41 @@
  */
 
 import * as rt from 'io-ts';
+import { v5 } from 'uuid';
+import { IdFormat, JobType } from '../http_api/log_analysis/v1/id_formats';
 
 export const bucketSpan = 900000;
 
 export const categoriesMessageField = 'message';
 
 export const partitionField = 'event.dataset';
 
-export const getJobIdPrefix = (spaceId: string, sourceId: string) =>
-  `kibana-logs-ui-${spaceId}-${sourceId}-`;
+const ID_NAMESPACE = 'f91b78c0-fdd3-425d-a4ba-4c028fe57e0f';
 
-export const getJobId = (spaceId: string, logViewId: string, jobType: string) =>
-  `${getJobIdPrefix(spaceId, logViewId)}${jobType}`;
+export const getJobIdPrefix = (spaceId: string, sourceId: string, idFormat: IdFormat) => {
+  if (idFormat === 'legacy') {
+    return `kibana-logs-ui-${spaceId}-${sourceId}-`;
+  } else {
+    // A UUID is 36 characters but based on the ML job names for logs, our limit is 32 characters
+    // Thus we remove the 4 dashes
+    const uuid = v5(`${spaceId}-${sourceId}`, ID_NAMESPACE).replaceAll('-', '');
```

Comment:
Suggestion: You could use `uuidv5.DNS` instead of `ID_NAMESPACE`

Reply:
Since we're not hashing a URL/domain name that might be confusing and it's not a big overhead to use a custom one, I'll stick with that.